pkg/cognito/cognito.go

```diff
@@ -2,16 +2,19 @@ package cognito
 
 import (
 	"fmt"
+	"strings"
 
 	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
 	"github.com/aws/aws-sdk-go/service/cognitoidentityprovider"
 	"github.com/oslokommune/okctl/pkg/apis/okctl.io/v1alpha1"
 )
 
 // Cognito contains all required state for interacting
 // with the Cognito API
 type Cognito struct {
-	provider v1alpha1.CloudProvider
+	provider   v1alpha1.CloudProvider
+	usprovider v1alpha1.CloudProvider
```

Comment:
I find this abit confusing. Shouldn't it be enough to send a single provider but instance it with an US region? Or are they functionality-wise different? In that case maybe we should name them with their functionality, like `creationProvider` and the `deletionProvider`.

Reply:
Nope, it is not possible to switch the region in a provider as it is made now, without running init agian. This is why I opted to having two of them